File: classes01/python/datatypeConversion.py
import cv2
import numpy as np
# 读取一个文件
source=cv2.imread("../../data/images/sample.jpg")

scaleFactor=1/255.0

# 将unsigned int 转为float
source=np.float32(source)
source=source*scaleFactor

cv2.imshow("float",source)
# 将float转为unsigned char
source=source*(1.0/scaleFactor)
source=np.uint8(source)
cv2.imshow("unsigned char",source)

cv2.waitKey(0)

File: classes01/cpp/datatypeConversion.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(void)
{
    // 读取数据
    Mat source=imread("../../data/images/sample.jpg");

    // 在转变像值的时候的缩放因子
    double scale=1/255.0;
    double shift=0.0;

    // 将unsighed char转换为32位float
    source.convertTo(source,CV_32FC3,scale,shift);

    imshow("32bit",source);

    // 将float转换为unsigned char
    source.convertTo(source,CV_8UC3,1.0/scale,shift);

    imshow("unsigned char",source);

    waitKey(0);

    return EXIT_SUCCESS;
}